platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -164,9 +175,32 @@ class ConfigurationCacheProblems(
         }
     }
 
+    override fun forTask(task: Task): ProblemsListener {
+        val taskDegradationReasons = degradationController.degradationReasons
```

Comment:
❓ Are we sure all configuration logic has already executed by the time this is called for the 1st time?